.github/workflows/integration-test.yml

```
@@ -32,13 +32,12 @@ jobs:
     # To prevent this, this condition ensures that integration tests are run
     # in only ONE of the following cases:
     #
-    #   1. The event is NOT a pull_request.  This covers the case when the event
-    #      is a pull_request_target (i.e., a PR from a fork), as well as all
-    #      other cases listed in the "on" block at the top of this file.
+    #   1. The event is NOT a pull_request (it's a pull_request_target) and the base
+    #      repo and the head repo are different (i.e., the PR is from a fork).
     #   2. The event IS a pull_request AND the base repo and head repo are the
     #      same (i.e., the PR is NOT from a fork).
     #
-    if: github.event_name != 'pull_request' || github.event.pull_request.base.repo.full_name == github.event.pull_request.head.repo.full_name
+    if: (github.event_name == 'pull_request') == (github.event.pull_request.base.repo.full_name == github.event.pull_request.head.repo.full_name)
```

I think this could be done like:
```suggestion
    if: (github.event_name != 'pull_request') && github.event.pull_request.head.repo.fork
```
but I the logic as written is fine and I can't decide which I like better.

---

I definitely prefer `github.event.pull_request.head.repo.fork` over `github.event.pull_request.base.repo.full_name == github.event.pull_request.head.repo.full_name`! nice find!

---

I had to figure out how to not push containers from forks back in the day:
https://github.com/ASFHyP3/actions/blob/develop/.github/workflows/reusable-docker-ghcr.yml#L58

---

Oops! Nope. I pulled in this change, but it is indeed NOT the same. This must also run with it _is_ a pull request _not_ from a fork.

---

that should be false-false though right, so and would evaluate to true?